submissions/sofiichuk/frogger-task/app.js

```diff
@@ -0,0 +1,147 @@
+    y: 385
+};
+
+let count = 0;
+
+// ===========================================================
+
+const score = document.createElement('h3');
+score.textContent = 'your score is: 0';
+document.body.append(score);
+
+// ===========================================================
+
+const Enemy = function (x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png'
+};
+
+Enemy.prototype.update = function (dt) {
+    if (this.x <= playingField.width) {
+        this.x = this.speed * dt + this.x;
+    } else {
+        this.x = 0;
+    }
+    this.detectCollision();
+};
+
+Enemy.prototype.detectCollision = function () {
+    if (player.x <= this.x + character.width &&
+        player.x + character.width >= this.x &&
+        player.y <= this.y + character.height &&
+        player.y + character.height >= this.y) {
+        player.reset()
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// ===========================================================
+
+const Player = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.moveSideways = cell.width;
+    this.moveAhead = cell.height;
+    this.sprite = 'images/char-boy.png';
+};
+
+Player.prototype.update = function () {
+    this.scored();
+};
+
+Player.prototype.scored = function () {
+    if (this.y < river.height) {
+        count++;
+        score.textContent = `your score is: ${count}`;
+        this.x = initialPlayerPosition.x;
+        this.y = initialPlayerPosition.y;
+    }
+};
+
+Player.prototype.reset = function () {
+    score.textContent = 'wasted';
+    this.x = initialPlayerPosition.x;
+    this.y = initialPlayerPosition.y;
+};
+
+Player.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.handleInput = function (key) {
+    switch (key) {
+        case 'up':
+            this.y -= this.moveAhead;
+            if (this.y < 0) {
+                this.y = 0;
+            }
+            break;
+        case 'down':
+            this.y += this.moveAhead;
+            if (this.y > initialPlayerPosition.y) {
+                this.y = initialPlayerPosition.y;
+            }
+            break;
+        case 'left':
+            this.x -= this.moveSideways;
+            if (this.x < 0) {
+                this.x = 0;
+            }
+            break;
+        case 'right':
+            this.x += this.moveSideways;
+            if (this.x > playingField.width - cell.width) {
+                this.x = playingField.width - cell.width;
+            }
+            break;
+    }
+};
+
+// ===========================================================
+
+const player = new Player(initialPlayerPosition.x, initialPlayerPosition.y);
+
+// const Enemy = function (x, y, speed)
+const enemy1 = new Enemy(0, 225, 100);
+const enemy2 = new Enemy(0, 142, 200);
+const enemy3 = new Enemy(0, 60, 150);
```

Comment:
Mystery numbers. Item 3 (all subitems) of the requirements.
Also not DRY. Imagine you are tasked to add 10 more enemies. Minimize the effort.

Reply:
no more mystery numbers.
no mo repetitions.
p.s. also fixed two bugs that had been found during UAT.
p.p.s. please, review .this once again